internal/email/email.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package email provides the email utilities for minder
+package email
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/google/uuid"
+
+	"github.com/stacklok/minder/internal/events"
+)
+
+const (
+	// TopicQueueInviteEmail is the topic for sending invite emails
+	TopicQueueInviteEmail = "invite.email.event"
+)
+
+// Service is the email service interface
+type Service interface {
+	SendEmail(ctx context.Context, to, subject, bodyHTML, bodyText string) error
+}
+
+// MailEventPayload is the event payload for sending an invitation email
+type MailEventPayload struct {
+	Code           string `json:"code"`
+	ProjectDisplay string `json:"project"`
+	SponsorDisplay string `json:"sponsor"`
+	Role           string `json:"role"`
+	Email          string `json:"email"`
+}
+
+// MailManager is the email manager
+type MailManager struct {
+	client Service
+}
+
+// NewMailManager creates a new mail manager
+func NewMailManager(client Service) *MailManager {
```

Comment:
I would call this something like `MailEventConsumer` or `MailEventHandler` to make its purpose more clear.

Reply:
I did not like this name too 😄 thanks! 👍

Reply:
Fixed 👍